File: internal/domain/message.go
package domain

import "time"

type Message struct {
	ID               *int
	ChatID           *int
	SenderID         *int
	SenderName       *string
	AuthorName       *string
	OriginalID       *int
	ParentID         *int
	ParentSenderName *string
	ParentText       *string
	Text             *string
	IsViewed         *bool
	SentAt           *time.Time
}

type MessageView struct {
	MessageID *int
	UserID    *int
}

File: web/src/models/message.ts
export interface Message {
    id: number
    chatId: number
    senderId: number
    senderName: string
    authorName: string
    originalId: number
    parentId: number
    parentSenderName: string
    parentText: string
    text: string
    sentAt: string
    isViewed: boolean
}
